index.js

```diff
@@ -112,7 +115,9 @@ var publicAgent = {
 
     common.utils.getInstanceId(function(err, instanceId) {
       if (err) {
-        logger.info('Unable to retrieve GCE instance id.', err);
+        if (err.code !== 'ENOTFOUND') {
+          logger.warn('Unable to retrieve GCE instance id.', err);
+        }
```

Comment:
I can be fine with checking the return code from getInstanceId, but we should document this in the API. Also, either you need a comment here explaining what `ENOTFOUND` implies, or make the API return a more descriptive error.